.gitignore

```diff
@@ -103,6 +103,7 @@ outputs/
 
 *.a
 *.o
+*.o.tmp
```

Comment:
Oh oops sorry this snuck in here. Clang build temp files like this, which I noticed when I started developing with clang, and so I added this line while working. They are deleted by Clang after the build is complete, but if you run `git status` while a build is in progress (say in another terminal), they clutter the output.

I had meant to PR this separately; let me know if I still should.